source/plugins/ruby/KubernetesApiClient.rb

```diff
@@ -943,6 +946,7 @@ def watch(resource_name, options = {})
         }
         http_headers = {
           Authorization: "Bearer " + getTokenStr,
+          "User-Agent": @@userAgent,
```

Comment:
can you please check in AKS API telemetry for your clusters and see User-Agent value being emitted for kube api calls (watch and non-watch) APIs?